### PR TITLE
Interfaces 338 refactor operator store

### PR DESCRIPTION
**Pull request by axelrod-blox**
INTERFACES-238: refactored operator store

### CODE REVIEW

**Comment by axelrod-blox** [comment]
don't merge until I test it by myself

**Comment by axelrod-blox** [comment]
- [ ] add operator
- [ ] remove operator
- [ ] withdraw operator balance
- [ ] Increase Fee
- [ ] Decrease Fee
- [ ] Cancel fee
- [ ] Add new validator (Keystore)
- [ ] Add new validator (KeyShare)
- [ ] Added second validator

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/DeclareFee.tsx [hunk 1]
```diff
@@ -16,34 +15,41 @@ import { useStores } from '~app/hooks/useStores';
...
         saveInLocalStorage('expired_operators', JSON.stringify(savedOperator));
       }
     }
-    getCurrentState();
     setIsLoading(false);
   };
 
+  useEffect(() => {
+    getCurrentState();
+  }, [JSON.stringify(operatorFeeData)]);
```
Comment (sumbat-ssvlabs) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/DeclareFee.tsx:51]: relying on property order might not be consistent, if possible, try to relay on something else

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx [hunk 2]
```diff
@@ -25,11 +25,10 @@ enum StepperProcessesSteps {
 }
 
 const ReactStepper = ({ step, registerButtonEnabled, subTextAlign, subText }: StepperProps) => {
-    const stores = useStores();
-    const operatorStore: OperatorStore = stores.Operator;
     const classes = useStyles({ step, registerButtonEnabled, subTextAlign });
+    const operatorFeeData = useAppSelector(getOperatorFeeData);
     // @ts-ignore
-    const time = (operatorStore.operatorApprovalBeginTime - operatorStore?.declaredOperatorFeePeriod) * 1000;
+    const time = (operatorFeeData.operatorApprovalBeginTime - operatorFeeData?.declaredOperatorFeePeriod) * 1000;
```
Comment (sumbat-ssvlabs) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx:31]: this one can result in NaN, please remove the @ts-ignore, and add a fallback like (operatorStore?.declaredOperatorFeePeriod || 0) or something else

**Review by sumbat-ssvlabs — APPROVED** [review]
approved with some comments

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx [hunk 3]
```diff
@@ -20,28 +19,33 @@ import { getStrategyRedirect } from '~app/redux/navigation.slice';
...
+  const operatorFeeData = useAppSelector(getOperatorFeeData);
+  const processOperatorId = useAppSelector(getOperatorProcessId);
 
   const cancelUpdateProcess = async () => {
-    if (!operatorStore.processOperatorId) return navigate(strategyRedirect);
-    const response = await operatorStore.cancelChangeFeeProcess({ operator, isContractWallet, dispatch });
+    if (!processOperatorId) return navigate(strategyRedirect);
+    const response = await cancelChangeFeeProcess({ operator, isContractWallet, dispatch });
     if (response) {
       // @ts-ignore
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx:46]: remove all ts-ignore here and in line 65 and 67

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx [hunk 4]
```diff
@@ -20,28 +19,33 @@ import { getStrategyRedirect } from '~app/redux/navigation.slice';
...
-      setFutureFee(operatorStore.operatorFutureFee);
+      await dispatch(fetchAndSetOperatorFeeInfo(operator.id));
+      setFutureFee(operatorFeeData.operatorFutureFee);
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx:48]: make return in fetchAndSetOperatorFeeInfo to get response (operatorFeeData will be in object.payload)
don't use outer operatorFeeData from useSelect

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx [hunk 5]
```diff
@@ -24,18 +22,13 @@ import { getStrategyRedirect } from '~app/redux/navigation.slice';
...
-};
+import {
+  clearOperatorFeeInfo,
+  fetchAndSetOperatorFeeInfo,
+  getFeeIncreaseAndPeriods, getMaxOperatorFeePerYear,
+  getOperatorFeeData,
+  getOperatorProcessId
+} from '~app/redux/operator.slice.ts';
 
 // eslint-disable-next-line no-unused-vars
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx:33]: remove eslint-disable

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx [hunk 6]
```diff
@@ -2,11 +2,9 @@ import { useEffect, useState } from 'react';
 import Grid from '@mui/material/Grid';
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx:3]: remove observer

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx [hunk 7]
```diff
@@ -2,9 +2,7 @@
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx:2]: remove observer

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/PendingExpired.tsx [hunk 8]
```diff
@@ -2,9 +2,7 @@
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/PendingExpired.tsx:2]: remove observer

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/PendingExpired.tsx [hunk 9]
```diff
@@ -2,9 +2,7 @@
...
 import Grid from '@mui/material/Grid';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/PendingExpired.tsx:3]: use Grid from atomicComponents

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx [hunk 10]
```diff
@@ -2,10 +2,10 @@
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx:1]: observer

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx [hunk 11]
```diff
@@ -2,10 +2,10 @@
...
 import Grid from '@mui/material/Grid';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx:2]: Grid from atomicComponents

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx [hunk 12]
```diff
@@ -4,12 +4,11 @@ import Grid from '@mui/material/Grid';
 import Typography from '@mui/material/Typography';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx:4]: remove observer
use Grid from atomicComponents

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx [hunk 13]
```diff
@@ -35,11 +35,11 @@ const UpdateFeeState = ({ operatorId }: { operatorId?: string }) => {
 
   const getState = async () => {
     // @ts-ignore
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx:37]: remove ts-ignore

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/WaitingPeriod.tsx [hunk 14]
```diff
@@ -3,30 +3,28 @@ import { observer } from 'mobx-react';
 import Grid from '@mui/material/Grid';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/WaitingPeriod.tsx:3]: remove observer
use Grid from atomicComponents

File: src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx [hunk 15]
```diff
@@ -1,41 +1,41 @@
 import { useEffect, useState } from 'react';
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx:2]: remove observer
use Grid from atomicComponents

File: src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx [hunk 16]
```diff
@@ -1,41 +1,41 @@
...
 
   useEffect(() => {
     getCurrentState();
   }, []);
 
   const getCurrentState = async () => {
 
-    await operatorStore.syncOperatorFeeInfo(operator.id);
-    if (operatorStore.operatorApprovalBeginTime && operatorStore.operatorApprovalEndTime && operatorStore.operatorFutureFee) {
+    await dispatch(fetchAndSetOperatorFeeInfo(operator.id));
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx:32]: use returned value from fetchAndSetOperatorFeeInfo

File: src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx [hunk 17]
```diff
@@ -4,18 +4,18 @@ import { observer } from 'mobx-react';
 import { useStyles } from '~app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/SelectOperators.styles';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx:4]: remove observer
use Grid from atomicComponents

File: src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx [hunk 18]
```diff
@@ -1,4 +1,4 @@
-import  { useEffect, useRef, useState } from 'react';
+import { useEffect, useRef, useState } from 'react';
 import _ from 'underscore';
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx:3]: remove observer

File: src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx [hunk 19]
```diff
@@ -95,30 +91,34 @@ const FirstSquare = ({ editPage, clusterSize, setClusterSize, clusterBox }: {
     setOperatorsPagination(response.pagination);
   };
 
-  const selectOperator = (e: any, operator: IOperator) => {
+  const selectOperatorHandling = (e: any, operator: IOperator) => {
     // @ts-ignore
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx:95]: remove ts-ignore

File: src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx [hunk 20]
```diff
@@ -1,29 +1,30 @@
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx:1]: remove observer
use Grid from atomicComponents

File: src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx [hunk 21]
```diff
@@ -45,11 +46,10 @@ const SetOperatorFee = () => {
   }, [error.shouldDisplay, userInput]);
 
   const moveToSubmitConfirmation = () => {
-    const operatorWithFee = operatorStore.newOperatorKeys;
-    // @ts-ignore
-    operatorWithFee.fee = parseFloat(userInput) || 0;
-    operatorStore.setOperatorKeys(operatorWithFee);
-    navigate(config.routes.SSV.OPERATOR.CONFIRMATION_PAGE);
+    console.log(operatorRawData);
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx:49]: remove log

File: src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx [hunk 22]
```diff
@@ -45,11 +46,10 @@ const SetOperatorFee = () => {
...
+    operatorRawData.fee = parseFloat(userInput) || 0;
+    console.log('inNavigate');
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx:51]: remove log

File: src/app/components/applications/SSV/ValidatorSuccessScreen/ValidatorSuccessScreen.tsx [hunk 23]
```diff
@@ -1,34 +1,32 @@
-import  { useRef, useState } from 'react';
+import { useRef, useState } from 'react';
 import { observer } from 'mobx-react';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/ValidatorSuccessScreen/ValidatorSuccessScreen.tsx:2]: remove observer
use Grid from atomicComponents

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/DeclareFee.tsx (hunk 1 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/DeclareFee.tsx:51]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx (hunk 2 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx:31]: +

File: src/app/redux/operator.slice.ts [hunk 24]
```diff
@@ -0,0 +1,153 @@
...
+      state.operatorApprovalEndTime = 0;
+      state.operatorFutureFee = 0;
+    }
+  },
+  extraReducers: (builder) => {
+    builder.addCase(fetchAndSetOperatorValidatorsLimit.fulfilled, (state, action: { payload: number }) => {
+      state.operatorValidatorsLimit = action.payload;
+    });
+    builder.addCase(fetchAndSetOperatorFeeInfo.fulfilled, (state, action) => {
+      if (action.payload) {
```
Comment (IlyaVi) [inline src/app/redux/operator.slice.ts:107]: why you check it there is payload?

File: src/services/operatorContract.service.ts [hunk 25]
```diff
@@ -47,7 +88,7 @@ const updateOperatorAddressWhitelist = async ({ operator, address, isContractWal
     const operatorAfter = await getOperator(operator.id);
     return operatorAfter.address_whitelist;
   };
-
+  console.log('1');
```
Comment (IlyaVi) [inline src/services/operatorContract.service.ts:91]: remove log

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx (hunk 3 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx:46]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx (hunk 10 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx:1]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx (hunk 11 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/Stepper.tsx:2]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx (hunk 12 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx:4]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx (hunk 13 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/UpdateFeeState.tsx:37]: +

File: src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx (hunk 15 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx:2]: +

File: src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx (hunk 17 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx:4]: +

File: src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx (hunk 18 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx:3]: +

File: src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx (hunk 19 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/RegisterValidatorHome/components/SelectOperators/components/FirstSquare/FirstSquare.tsx:95]: it's make a error, I will refactor all this component soon

File: src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx (hunk 20 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx:1]: +

File: src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx (hunk 21 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx:49]: +

File: src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx (hunk 22 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx:51]: +

File: src/app/components/applications/SSV/ValidatorSuccessScreen/ValidatorSuccessScreen.tsx (hunk 23 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/ValidatorSuccessScreen/ValidatorSuccessScreen.tsx:2]: +

File: src/app/redux/operator.slice.ts (hunk 24 above)
Comment (axelrod-blox) [reply src/app/redux/operator.slice.ts:107]: +

File: src/services/operatorContract.service.ts (hunk 25 above)
Comment (axelrod-blox) [reply src/services/operatorContract.service.ts:91]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/PendingExpired.tsx (hunk 8 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/PendingExpired.tsx]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx (hunk 7 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/components/CancelFee.tsx]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx (hunk 6 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx (hunk 5 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/UpdateFee/UpdateFee.tsx]: +

File: src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx (hunk 4 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditFeeFlow/CancelUpdateFee/CancelUpdateFee.tsx:47]: +

File: src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx (hunk 16 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/components/OperatorBox/OperatorBox.tsx:28]: done

**Pull request merged**